lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -172,6 +172,22 @@ static bool isNoopCast(Operation *op) {
   return false;
 }
 
+/// Return the word (e.g. "reg") in Verilog to declare the specified thing.
+static StringRef getVerilogDeclWord(Operation *op) {
+  if (isa<RegOp>(op))
+    return "reg";
+  if (isa<WireOp>(op) || isa<MergeOp>(op))
+    return "wire";
+
+  // Interfaces instances use the name of the declared interface.
+  if (auto interface = dyn_cast<InterfaceInstanceOp>(op))
+    return interface.getInterfaceType().getInterface().getValue();
+
+  if (isa<RTLModuleOp>(op->getParentOp()))
+    return "wire";
```

Comment:
The patch looks good, but I'm curious about this part: why default to wire at the top level of a module?

Reply:
Sorry, but what's the alternative? Have you seen my [comment](https://github.com/llvm/circt/issues/439#issuecomment-779001339) about `logic` not working at the RTL module level?